File: 1. Two Sum/solution.cpp
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
      unordered_map<int, int> match = {};
      for(int index = 0; 
          index < nums.size();
          index++){
        int value = nums.at(index);
        if(match.find(value) != match.end()){
          nums.clear();
          nums.push_back(match.at(value));
          nums.push_back(index);
          return nums;
          //int result[] = {match.at(value), index};
          //return vector<int>(result, result+2);
        }else{
          match[target - value] = index;
        }
      }
      return nums;
    }
};

File: 1. Two Sum/solution.rs
use std::collections::HashMap;
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut matches = HashMap::new();
        for (index, num) in (0..).zip(nums.into_iter()){
            if matches.contains_key(&num) {
                let mut nums = Vec::new();
                nums.push(matches.get(&num).unwrap().to_owned());
                nums.push(index);
                return nums;
            }
            matches.insert(target - num, index);
        }
        return vec![];
    }
}
